src/bit-systems/scene-preview-camera-system.ts

```diff
@@ -0,0 +1,119 @@
+import { defineQuery, enterQuery, exitQuery } from "bitecs";
+import { HubsWorld } from "../app";
+import { ScenePreviewCamera } from "../bit-components";
+import { Quaternion, Vector3, Euler, Camera } from "three";
+import { CAMERA_MODE_INSPECT, CAMERA_MODE_SCENE_PREVIEW, CameraSystem } from "../systems/camera-system";
+import { setMatrixWorld } from "../utils/three-utils";
+import { getStreamerCamera } from "../components/scene-preview-camera";
+import { lerp } from "three/src/math/MathUtils";
+
+const startPositions = new Map<number, Vector3>();
+const startRotations = new Map<number, Quaternion>();
+const targetPositions = new Map<number, Vector3>();
+const targetRotations = new Map<number, Quaternion>();
+const startTimes = new Map<number, number>();
+const backwards = new Map<number, boolean>();
+const ranOnePass = new Map<number, boolean>();
+const newRot = new Quaternion();
+
+const updatePreviewCamera = (eid: number, previewCamera: Camera, cameraSystem: CameraSystem) => {
+  cameraSystem.mode = CAMERA_MODE_SCENE_PREVIEW;
+  const streamerCamera = getStreamerCamera();
+  if (streamerCamera) {
+    setMatrixWorld(previewCamera, streamerCamera.matrixWorld);
+    // Move camera forward just a bit so that we don't see the avatar's eye cylinders.
+    previewCamera.translateZ(-0.1);
+    previewCamera.matrixNeedsUpdate = true;
+  } else {
+    const startTime = startTimes.get(eid)!;
+    const duration = ScenePreviewCamera.duration[eid];
+    let t = (performance.now() - startTime) / (1000.0 * duration);
```

Comment:
`performance.now()` may be replaced with `world.time.elapsed` where `world` is `HubsWorld`.

Reply:
yep, that makes sense. Thanks.